### DISCUSSION TITLE
Does my signal space volume influence field fluctuations?

### DISCUSSION THREAD

**Discussion by iason-saganas** [NIFTy-PPL/NIFTy#30] [Q&A] [answered]
Hi! First of all, thank you for developing and maintaining `NIFTy`. I am finishing a project in Non-JAX, regular `NIFTy`, so I hope I am at the right place for asking this question.

### Starting situation

I can initialize a simple correlated field e.g. as follows: 

```
import nifty8 as ift

# Arguments of the correlated field model
args_cfm = {
    'offset_mean': 0,
    'offset_std': None,
    'fluctuations': (.25, .12),
    'loglogavgslope': (-4, 1e-16),
    'asperity': None,
    'flexibility': None,
}

s = ift.SimpleCorrelatedField(target=x, **args_cfm)
```

where `x` is the signal space `x=ift.RGSpace(n_pix, distances=pxl_size)`. 

From a math POV, I understand that the pointwise variance of my field is given by: `<s_x*s_x>=fluc^2`, where `fluc` stands for the fluctuations parameter. 

### My confusion

In the source code for the simple correlated field (ressource [1], see links down below), the amplitude power spectrum is constructed and in the following line 116, the apply method of the class `_Normalization` is called: 

```
a = _Normalization(pspace, 0) @ a
```

This function traces back to the source code for the correlated field (ressource [2]), line 158. I copy-paste the doc-string of this class, which is the source of my confusion: 

```
class _Normalization(Operator):
    """Exponentiate the logarithmic power spectrum, normalize by the sum over
    all modes and return the square root of the "normalized" power spectrum.

    Notes
    -----
    The operator does not perform a proper normalization as it does not account
    for changes in position space volume. This leads to an additional factor of
    `1 / \\sqrt{totvol}` being introduced into the result with `totvol`
    referring to the total volume in position space. The exact value of the
    additional factor stems from the fact that the volume in harmonic space is
    solely dependent on the distances in position space. Thus, if the position
    spaces is enlarged without changing its distances, the volume in harmonic
    space is kept constant. Doubling the number of pixels though also doubles
    the number of harmonic modes with each then occupy a smaller volume. This
    linear decrease in per pixel volume in harmonic space is not captured by
    just summing up the modes.
    """
```

According to this, there is an additional factor of `1/sqrt(totvol)` introduced upon "normalization". My question is: Does this factor propagate into the final result, i.e. _does the powerspectrum and therefore also the set field variance, have a dependance on the volume of my signal space?_ 

I tried to answer this question myself: In line 125 of the simple correlated field source code [1]: 

```
a = a.scale(target.total_volume)
```

Since this just represents a multiplication with the volume of the signal space `x`, which is `totvol` (_correct_?), this would cancel out the dependance on the space volume as introduced by `_Normalization`, _if_ the latter were `1/totvol`.
 But since it is `1/sqrt(totvol)` according to the doc-string, this leads me to believe that indeed a dependence on the volume is propagated into the final powerspectrum. The powerspectrum is squared only at the end ([1], line 134), after already having scaled with the space volume. So, if my argumentation is correct, the final power spectrum has a factor of `(1/sqrt(totvol)*totvol)^2=totvol`. 

Is this right? If yes, would I need to adjust my fluctuations parameter to be `fluct' = fluct/sqrt(totvol)`, such that `sqrt(totvol)^2` will get cancelled out by `a.scale(target.total_volume)` and I get my originally desired `fluc` value?

### Relevant links 
- [[1] Gitlab source code correlated field simple](https://gitlab.mpcdf.mpg.de/ift/nifty/-/blob/NIFTy_8/src/library/correlated_fields_simple.py)
- [[2] Gitlab source code correlated field](https://gitlab.mpcdf.mpg.de/ift/nifty/-/blob/NIFTy_8/src/library/correlated_fields.py)

**Comment by Edenhofer** [comment, marked as answer]
The fluctuation parameter configures the per-pixel standard deviation of the correlated field. There is no need for additional volume-dependent adjustments if that is what you want to configure here.

In the correlated field model in NIFTy/NIFTy.re you will encounter a lot of scalings with volumes. Their purpose is to make the parameters of the correlated field model easier to interpret for the user - easier in the sense that a user does not need to think about harmonic spaces too much. Most of these scalings concern the zero-mode and/or the harmonic transformation and are not relevant for the fluctuation.
